stdlib/public/Observation/Sources/Observation/Locking.swift

```diff
@@ -0,0 +1,85 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+
+@_silgen_name("_swift_observation_lock_size")
+func _lockSize() -> Int
+
+@_silgen_name("_swift_observation_lock_init")
+func _lockInit(_: UnsafeRawPointer)
+
+@_silgen_name("_swift_observation_lock_lock")
+func _lockLock(_: UnsafeRawPointer)
+
+@_silgen_name("_swift_observation_lock_unlock")
+func _lockUnlock(_: UnsafeRawPointer)
+
+@available(SwiftStdlib 5.9, *)
+struct ManagedCriticalState<State> {
```

Comment:
Crucial nit: In the stdlib, we strictly follow [the leading underscore rule](https://github.com/apple/swift/blob/main/docs/StandardLibraryProgrammersManual.md#the-leading-underscore-rule) when naming internal types and functions.

> All APIs that aren't part of the stdlib's official public API must include at least one underscored component in their fully qualified names. This includes symbols that are technically declared public but that aren't considered part of the public stdlib API, as well as @usableFromInline internal, plain internal and [file]private types and members.
> ...
> This rule ensures we don't accidentally clutter the public namespace with @usableFromInline things (which could prevent us from choosing the best names for newly public API later), and it also makes it easy to see at a glance if a piece of stdlib code uses any non-public entities.

(Things declared purely internal have a remarkable tendency to become `@usableFromInline` or `@inlinable` before the release ships, usually as a side effect of performance work.)